Lib/enum.py

```diff
@@ -916,45 +1101,106 @@ def _generate_next_value_(name, start, count, last_values):
         """
         if not count:
             return start if start is not None else 1
-        for last_value in reversed(last_values):
-            try:
-                high_bit = _high_bit(last_value)
-                break
-            except Exception:
-                raise TypeError('Invalid Flag value: %r' % last_value) from None
+        last_value = max(last_values)
+        try:
+            high_bit = _high_bit(last_value)
+        except Exception:
+            raise TypeError('Invalid Flag value: %r' % last_value) from None
         return 2 ** (high_bit+1)
 
     @classmethod
-    def _missing_(cls, value):
+    def _iter_member_by_value_(cls, value):
         """
-        Returns member (possibly creating it) if one can be found for value.
+        Extract all members from the value in definition (i.e. increasing value) order.
         """
-        original_value = value
-        if value < 0:
-            value = ~value
-        possible_member = cls._create_pseudo_member_(value)
-        if original_value < 0:
-            possible_member = ~possible_member
-        return possible_member
+        for val in _iter_bits_lsb(value):
+            member = cls._value2member_map_.get(val)
+            if member is not None:
+                yield member
```

Comment:
can the None filtering be removed by masking the value?

```python3
for val in _iter_bits_lsb(value & cls._flag_mask_):
    yield cls._value2member_map_.get(val)
```

Reply:
It can, and now is.